### PR TITLE
infra(tsconfig): noImplicitAny

### LINKED ISSUE #1837: Turn on noImplicitAny in tsconfig
https://github.com/faker-js/faker/pull/651 partially activated strict mode, this flag was omitted at that time and should probably be turned on.

https://github.com/faker-js/faker/blob/c46c53948efaf9ccb2597f36954128f914b73179/tsconfig.json#L15

Related issues:

- https://github.com/faker-js/faker/pull/651
- https://github.com/faker-js/faker/issues/1834
- https://github.com/faker-js/faker/issues/1835
- https://github.com/faker-js/faker/issues/1836
- https://github.com/faker-js/faker/issues/1838

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes #1837

- #1837

### CODE REVIEW

**Comment by ST-DDT** [comment]
I'm really unhappy with typescripts behavior here and I hope this gets better in a future version.

File: scripts/generate-locales.ts [hunk 1]
```diff
@@ -23,10 +23,8 @@ import {
 } from 'node:fs';
 import { dirname, resolve } from 'node:path';
 import { fileURLToPath } from 'node:url';
-import type { Options } from 'prettier';
-import { format } from 'prettier';
-import options from '../.prettierrc.js';
 import type { LocaleDefinition, MetadataDefinition } from '../src/definitions';
+import { formatMarkdown, formatTypescript } from './apidoc/format.js';
```
Comment (xDivisionByZerox) [inline scripts/generate-locales.ts]: Is the `*.js` correct here?

**Comment by xDivisionByZerox** [comment]
Are you talking about `Object.keys` not returning the type `(keyof T)[]`? If thats the case, there is [a PR comment in the typescript repository](https://github.com/Microsoft/TypeScript/pull/12253#issuecomment-263132208) that explains why it is this way.

We could make an internal helper function that is "correctly" typed:

```ts
function keys<T extends {}>(obj: T): (keyof T)[] {
  return Object.keys(obj) as (keyof T)[];
}
```

That way, we only have to cast in one place.

File: scripts/apidoc/format.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import type { Options } from 'prettier';
 import { format } from 'prettier';
+// @ts-expect-error: No types available.
 import prettierConfig from '../../.prettierrc.js';
```
Comment (xDivisionByZerox) [inline scripts/apidoc/format.ts]: Would we get type information if this was a JSON file?

File: scripts/generate-locales.ts (hunk 1 above)
Comment (ST-DDT) [reply scripts/generate-locales.ts]: Fixed.

File: scripts/apidoc/format.ts (hunk 2 above)
Comment (ST-DDT) [reply scripts/apidoc/format.ts]: I worked around the problem.

File: .prettierrc.d.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import type { Options } from 'prettier';
+
+declare const options: Options;
+export = options;
```
Comment (ST-DDT) [inline .prettierrc.d.ts:4]: Or is that better:

```suggestion
declare module './.prettierrc' {
  const options: Options;
  export default options;
}
```

Comment (xDivisionByZerox) [reply .prettierrc.d.ts:4]: But now we have an additional file to make the config work. Not sure about that.

Comment (ST-DDT) [reply .prettierrc.d.ts:4]: I just tested JSON as well, but there seems to be issues with string vs union types.

grafik

Should I revert to the `ts-expect-error` then?

Comment (xDivisionByZerox) [reply .prettierrc.d.ts:4]: If JSON does not work out of the box I'm fine with the custom definition file, for now. We might want to look into this at a later point again. Or maybe, prttier will allow TS configuration files, in the future.

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**